product-scenarios/3-securing-an-api-with-different-security-protocols/3.2-api-developers-want-to-secure-their-api-using-an-user-access-control-machanism/3.2.1-api-developer-wants-to-secure-api-resources-using-user-roles/src/test/java/org/wso2/am/scenario/tests/api/secure/userRoles/SecureUsingUserRolesNegativeTestCase.java

```diff
@@ -158,6 +262,55 @@ public void testScopeWithDuplicateKey() throws Exception {
                 "Error in scope creation with duplicate key : " + ITEM_VIEW);
     }
 
+    private void updateAPISwagger(String role, JSONObject swaggerJson) throws Exception {
+        swaggerJson.getJSONObject("x-wso2-security").getJSONObject("apim").getJSONArray("x-wso2-scopes").
+                getJSONObject(0).remove("roles");
+
+        swaggerJson.getJSONObject("x-wso2-security").getJSONObject("apim").getJSONArray("x-wso2-scopes").
+                getJSONObject(0).accumulate("roles", role);
+
+        String updatedPayload = swaggerJson.toString();
+
+        HttpResponse updateResponse = apiPublisher.updateResourceOfAPI(SUPER_USER, apiName, apiVersion, updatedPayload);
+        verifyResponse(updateResponse);
+    }
+
+    private void createNewApplicationAndSubscribe(String user, String password) throws Exception {
+        apiStore.login(user, password);
+        apiStore.addApplication(applicationName, APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED, "", "");
+
+        SubscriptionRequest subscriptionRequest = new SubscriptionRequest(apiName, SUPER_USER);
+        subscriptionRequest.setApplicationName(applicationName);
+        apiStore.subscribe(subscriptionRequest);
+    }
+
+    private String generateKeysAndAccessToken(String user, String password, String requiredScope) throws Exception {
+        //Generate production token and invoke with that
+        APPKeyRequestGenerator generateAppKeyRequest = new APPKeyRequestGenerator(applicationName);
+        String responseString = apiStore.generateApplicationKey(generateAppKeyRequest).getData();
+        JSONObject jsonResponse = new JSONObject(responseString);
+
+        // get Consumer Key and Consumer Secret
+        String consumerKey = jsonResponse.getJSONObject("data").getJSONObject("key").getString("consumerKey");
+        String consumerSecret = jsonResponse.getJSONObject("data").getJSONObject("key").getString("consumerSecret");
+
+        URL tokenEndpointURL = new URL(gatewayHttpsURL + "/token");
+        HttpResponse response;
+        String requestBody;
+        JSONObject accessTokenGenerationResponse;
+
+        //Obtain user access token for user
+        if (requiredScope.isEmpty()) {
```

Comment:
i think there wont be any change if you send a token req with &scope= 
In that case we can omit this if condition and can only have null check. please check